client/hotrod-client/src/main/java/org/infinispan/client/hotrod/impl/transport/netty/ChannelPool.java

```diff
@@ -142,41 +147,41 @@ private void createAndInvoke(ChannelOperation callback) {
                assert currentActive >= 0;
                int currentCreated = created.decrementAndGet();
                assert currentCreated >= 0;
+               if (trace) log.tracef(throwable, "Channel could not be created, created = %d, active = %d",
+                                     currentCreated, currentActive);
                callback.cancel(address, throwable);
             } else {
                callback.invoke(channel);
             }
          });
       } catch (Throwable t) {
-         active.decrementAndGet();
-         created.decrementAndGet();
+         int currentActive = active.decrementAndGet();
+         assert currentActive >= 0;
```

Comment:
why are we adding assert here instead of just firing the logs and exceptions ?

Reply:
Unlike `release` or `releaseClosedChannel`, here the client is guaranteed to be catching the `AssertionError`, because it needs a channel to actually execute the operation. However, this code path is really uncommon, so I doubt it will ever trigger the assert. I'll update it to just log a warning, like the others.

Reply:
Changed to log warnings instead